packages/front-end/components/PowerCalculation/stats.ts

```diff
@@ -340,3 +356,294 @@ export function powerMetricWeeks(
   };
   return results;
 }
+
+/*******************/
+export function calculatePriorMean(
+  priorMeanRel: number,
+  mean: number,
+  relative: boolean
+): number {
+  return relative ? priorMeanRel : priorMeanRel * Math.abs(mean);
+}
+
+export function calculatePriorVariance(
+  priorVarianceRel: number,
+  mean: number,
+  relative: boolean
+): number {
+  return relative
+    ? priorVarianceRel
+    : priorVarianceRel * Math.pow(Math.abs(mean), 2);
+}
+
+function calculatePriorMeanSpecified(
+  metric: MetricParams,
+  relative: boolean
+): number {
+  const metricMean = getMetricMean(metric);
+  return calculatePriorMean(metric.priorLiftMean, metricMean, relative);
+}
+
+function calculatePriorVarianceSpecified(
+  metric: MetricParams,
+  relative: boolean
+): number {
+  const metricMean = getMetricMean(metric);
+  return calculatePriorVariance(
+    Math.pow(metric.priorLiftStandardDeviation, 2),
+    metricMean,
+    relative
+  );
+}
+
+function calculatePriorMeanDGP(
+  metric: MetricParams,
+  relative: boolean
+): number {
+  const metricMean = getMetricMean(metric);
+  return calculatePriorMean(metric.effectSize, metricMean, relative);
+}
+
+function calculatePriorVarianceDGP(
+  metric: MetricParams,
+  relative: boolean
+): number {
+  const metricMean = getMetricMean(metric);
+  const priorStandardDeviationDGP = 0;
+  return calculatePriorVariance(
+    Math.pow(priorStandardDeviationDGP, 2),
+    metricMean,
+    relative
+  );
+}
+
+// Function to estimate variance of tau hat conditional on tau
+function estimateTauHatVariance(
+  metric: MetricParams,
+  nPerVariation: number,
+  relative: boolean
+): number {
+  const s = powerStandardError(metric, nPerVariation, relative);
+  return Math.pow(s, 2);
+}
+
+// Function to calculate marginal variance of tau hat
+function getMarginalVarianceTauHat(
+  metric: MetricParams,
+  nPerVariation: number,
+  relative: boolean
+): number {
+  const priorVarianceDGP = calculatePriorVarianceDGP(metric, relative);
+  const tauHatVariance = estimateTauHatVariance(
+    metric,
+    nPerVariation,
+    relative
+  );
+  return tauHatVariance + priorVarianceDGP;
+}
+
+// Function to calculate posterior precision
+function getPosteriorPrecision(
+  metric: MetricParams,
+  nPerVariation: number,
+  relative: boolean
+): number {
+  const priorVarianceSpecified = calculatePriorVarianceSpecified(
+    metric,
+    relative
+  );
+  const tauHatVariance = estimateTauHatVariance(
+    metric,
+    nPerVariation,
+    relative
+  );
+  const properInt = metric.proper ? 1 : 0;
+  return 1 / tauHatVariance + properInt / priorVarianceSpecified;
+}
+
+// Function to calculate upper cutpoint
```

Comment:
Comment is out of date as upper/lower is parameterized in this fn.

Reply:
removed, thanks